stable_baselines/td3/td3.py

```diff
@@ -294,121 +294,121 @@ def learn(self, total_timesteps, callback=None, seed=None,
-                        obs = self.env.reset()
-                    episode_rewards.append(0.0)
+                for i in range(self.n_envs):
+                    assert action[i].shape == self.env.action_space.shape
+
+                obs, reward, done, info = self.env.step(rescaled_action)
+
+                for i in range(self.n_envs):
+                    # Store transition in the replay buffer.
+                    self.replay_buffer.add(prev_obs[i], action[i], reward[i], obs[i], float(done[i]))
+                    # Retrieve reward and episode length if using Monitor wrapper
+                    maybe_ep_info = info[i].get('episode')
+                    if maybe_ep_info is not None:
+                        ep_info_buf.extend([maybe_ep_info])
+                    if writer is not None:
+                        # Write reward per episode to tensorboard
+                        ep_reward = np.array([reward[i]]).reshape((1, -1))
+                        ep_done = np.array([done[i]]).reshape((1, -1))
+                        self.episode_reward = total_episode_reward_logger(self.episode_reward, ep_reward,
+                                                                        ep_done, writer, self.num_timesteps)
+
+                    if step % self.train_freq == 0:
+                        mb_infos_vals = []
+                        # Update policy, critics and target networks
+                        for grad_step in range(self.gradient_steps):
+                            # Break if the warmup phase is not over
+                            # or if there are not enough samples in the replay buffer
+                            if not self.replay_buffer.can_sample(self.batch_size) \
+                                    or self.num_timesteps < self.learning_starts:
+                                break
+                            n_updates += 1
+                            # Compute current learning_rate
+                            frac = 1.0 - step / total_timesteps
+                            current_lr = self.learning_rate(frac)
+                            # Update policy and critics (q functions)
+                            # Note: the policy is updated less frequently than the Q functions
+                            # this is controlled by the `policy_delay` parameter
+                            mb_infos_vals.append(
+                                self._train_step(step, writer, current_lr, (step + grad_step) % self.policy_delay == 0))
+                    
+                        # Log losses and entropy, useful for monitor training
+                        if len(mb_infos_vals) > 0:
+                            infos_values = np.mean(mb_infos_vals, axis=0)
+                    
+                    episode_rewards[-1] += reward[i]
+                    if done[i]:
+                        if self.action_noise is not None:
+                            self.action_noise.reset()
```

Comment:
same remark as before, I think you should have a action_noise object per env, maybe we need to create wrapper for that or modify the noise class to handle it better